File: client/bindings/nodejs/types/burn.ts
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import type { INativeToken } from '@iota/types';

/** A DTO for [`Burn`] */
export interface Burn {
    /** Aliases to burn */
    aliases?: string[];
    /** NFTs to burn */
    nfts?: string[];
    /** Foundries to burn */
    foundries?: string[];
    /** Amounts of native tokens to burn */
    nativeTokens?: INativeToken[];
}

File: client/bindings/java/lib/src/main/java/org/iota/types/Burn.java
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package org.iota.types;

import com.google.gson.JsonObject;

public class Burn extends AbstractObject {

    public Burn(JsonObject jsonObject) {
        super(jsonObject);
    }

    public Burn(String jsonObject) {
        super(jsonObject);
    }

}